soc/nxp/mcx/mcxe/mcxe31x/pinctrl_soc.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_NXP_MCXE31X_PINCTRL_SOC_H_
+#define ZEPHYR_SOC_NXP_MCXE31X_PINCTRL_SOC_H_
+
+#include <zephyr/dt-bindings/pinctrl/nxp-s32-pinctrl.h>
+#include <zephyr/sys/util.h>
+
+#include "../../../s32/common/siul2_pinctrl.h"
```

Comment:
I'd suggest moving this file to an NXP common directory. Furthermore, because the SIUl2 driver is now native and doesn't depend on MCUX or S32 SDK's, we should remove the S32 references from all places.
We can defer to another PR to not bloat this one, but I wanted to mention it since this is the first non-S32 platform that is sharing a lot of hw IPs with the S32.